test/integration/replicationcontroller/replicationcontroller_test.go

```diff
@@ -494,6 +495,45 @@ func TestSpecReplicasChange(t *testing.T) {
 	}
 }
 
+func TestLogarithmicScaleDown(t *testing.T) {
```

Comment:
This test is ensuring that the existing behavior somewhat still holds. That is good. We should run it with the feature gate enabled and disabled (I would remove the `Logarithmic` part from the test name)

Another test we could have is rather opposite: have all the pods run at roughly the same time, downscale, and see if some level of randomness holds. Do you think something like this is possible? Maybe we can run the same test X times and ensure that at least in one of them, the removed pod was not the absolute youngest.

In the test plan we also suggested emulating the story https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/2185-random-pod-select-on-replicaset-downscale#test-plan
We can do that in a follow up.

Reply:
> have all the pods run at roughly the same time, downscale, and see if some level of randomness holds ... Maybe we can run the same test X times

There is an inherent problem with randomness in that we can never be certain it will be the randomness we expect :) Something like this will introduce flakes, which can be minimized by increasing the number of test runs, but still always possible. Are there examples of other tests which take a similar approach?

What we really wish to show from such a test is that pods created on a similar logarithmic time scale have an equal chance of being chosen for downscaling. In other words, that their ranks are calculated properly. I believe the unit tests cover this level of detail sufficiently. 

The integration test is simply showing that the basic logic still works when removed from the vacuum of unit tests. I think that the user story you linked will make a good e2e, and I could actually add that to this PR (I don't see any reason to wait, and that will flesh out test cases here more comfortably). Wdyt?

Reply:
> I believe the unit tests cover this level of detail sufficiently.

I agree, but thanks for giving it a thought.

Still, modify the test to run for feature gate enabled and disabled.

Fine with me to add the 2e2 test for the user story in this PR.